### PR TITLE
pytest: add pytest markers specifying fork validity for tests + framework cleanup

### PR DESCRIPTION
**Pull request by danceratopz**
- [x] TBD Change fork from parameter to (autouse) fixture. Add global fixture?
  - [x] Don't forget the 1 4844 test with a Shanghai fork.
- [x] Implement `--forks-from London --forks-until Cancun`
  - [x] Don't run 4844 on London. Take the newest fork from the test case and the command-line options (on a per test case basis).
- [x] Add `fill` entry point.
- [x] Move `./tests/conftest.py` to `./src/pytest_plugins/test_filler.py` (keep all framework code in same place) ini: `-m test_filler`
- [x] Change default output directory from `out` to `fixtures`.
- [x] Add versions of solc and evm to pytest console output header
     Required by "[Getting Started](https://danceratopz.github.io/execution-spec-tests/getting_started/quick_start/)" screenshots.
- [x] Add tests fork range / valid marker logic
- [x] Consider how to change/better use forks in ethereum_test_forks to avoid hard-coded forks in `forks` plugin.

### CODE REVIEW

**Comment by danceratopz** [comment]
This fails with empty fork range; it should just not find any (valid) tests and exit without error:
```
fill fillers/eips/eip3651.py --until Merge
```

File: fillers/example/acl_example.py [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
-from ethereum_test_forks import Berlin, Fork, London, forks_from_until, is_fork
+from ethereum_test_forks import Fork, London, is_fork
 from ethereum_test_tools import AccessList, Account, Environment
 from ethereum_test_tools import Opcodes as Op
 from ethereum_test_tools import StateTestFiller, Transaction
 
 
-@pytest.mark.parametrize("fork", forks_from_until(Berlin, London))
+@pytest.mark.valid_from("Berlin")
+@pytest.mark.valid_until("London")
```
Comment (marioevz) [inline fillers/example/acl_example.py:14]: Amazing :D

**Review by marioevz — COMMENTED** [review]
A few comments. I have a branch that attempts to solve the fork transition tests issues. I'll open a PR to your branch and we can discuss about it :D

File: src/pytest_plugins/forks.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+    available_forks_help = textwrap.dedent(
+        f"""\
+        Available forks:
+        {", ".join(config.fork_names)}
+        """
+    )
+    dev_forks_help = textwrap.dedent(
+        "To run tests for a fork under active development, it must be "
+        "specified explicitly via --forks-until=FORK.\n"
```
[the remark this answers is not in the document]
Comment (danceratopz) [reply src/pytest_plugins/forks.py:88]: Yup, that's correct! I intended to add aliases to allow both `--forks-until` and `until` but then it didn't work.

File: src/pytest_plugins/forks.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    ]
+    valid_from = [
+        marker.args[0]
+        for marker in metafunc.definition.iter_markers(name="valid_from")
+    ]
+    valid_until = [
+        marker.args[0]
+        for marker in metafunc.definition.iter_markers(name="valid_until")
+    ]
+    if not valid_only and not valid_from:
```
[the remark this answers is not in the document]
Comment (danceratopz) [reply src/pytest_plugins/forks.py:228]: Yes, that's the intended behavior (as long as there are valid test cases from genesis of course).

File: src/pytest_plugins/forks.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        pytest.fail(
+            "The test function "
+            f"{metafunc.function.__name__} does not specify the first "
+            "evm fork for which it is valid. It must be marked with either "
+            "@pytest.mark.valid_only_for(FORK) or "
+            "@pytest.mark.valid_from(FORK)."
+        )
```
[the remark this answers is not in the document]
Comment (danceratopz) [reply src/pytest_plugins/forks.py:235]: PR looks great on first view, will take a closer look.

File: src/evm_transition_tool/__init__.py [hunk 5]
```diff
@@ -267,8 +278,8 @@ def version(self) -> str:
 
         return self.cached_version
 
-    def is_fork_supported(self, _: Fork) -> bool:
+    def is_fork_supported(self, fork: Fork) -> bool:
         """
         Returns True if the fork is supported by the tool
         """
-        return True
+        return fork().name() in self.help_string
```
[the remark this answers is not in the document]
Comment (danceratopz) [reply src/evm_transition_tool/__init__.py:285]: Would be nice if `evm` provided a better way to check and this implementation is a bit rough. But it is a big help to get the feedback directly!

**Review by marioevz — APPROVED** [review]
LGTM

**Pull request merged**